qemu/tests/nic_hotplug.py

```diff
@@ -115,84 +124,95 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
+                for nic in disable_nic_list:
+                    if guest_is_linux:
+                        ifname = utils_net.get_linux_ifname(s_session, nic["mac"])
+                        s_session.cmd_output_safe("ifconfig %s 0.0.0.0" % ifname)
+                    else:
+                        s_session.cmd_output_safe("ipconfig /release all")
+                    vm.set_link(nic.device_id, up=False)
+
+                error.context("Hotplug %sth '%s' nic named '%s'" % (nic_index,
+                                                                    nic_model,
+                                                                    nic_name))
+                hotplug_nic = vm.hotplug_nic(**nic_params)
+                error.context("Check if new interface gets ip address",
                               logging.info)
-                unplug_nic_index = random.randint(0, len(nic_hotplugged) - 1)
-                vm.hotunplug_nic(nic_hotplugged[unplug_nic_index].nic_name)
-                nic_hotplugged.pop(unplug_nic_index)
+                hotnic_ip = get_hotplug_nic_ip(
+                    vm,
+                    hotplug_nic,
+                    s_session,
+                    guest_is_linux)
+                if not hotnic_ip:
+                    raise error.TestFail("Hotplug nic can get ip address")
+                logging.info("Got the ip address of new nic: %s", hotnic_ip)
+
+                error.context("Ping guest's new ip from host", logging.info)
+                status, output = ping_hotplug_nic(host_ip_addr, hotplug_nic["mac"],
+                                                  s_session, guest_is_linux)
+                if status:
+                    err_msg = "New nic failed ping test, error info: '%s'"
+                    raise error.TestFail(err_msg % output)
+
+                error.context("Reboot vm after hotplug nic", logging.info)
+                # reboot vm via serial port since some guest can't auto up
+                # hotplug nic and next step will check is hotplug nic works.
                 s_session = vm.reboot(session=s_session, serial=True)
                 vm.verify_alive()
+                hotnic_ip = get_hotplug_nic_ip(
+                    vm,
+                    hotplug_nic,
+                    s_session,
+                    guest_is_linux)
+                if not hotnic_ip:
+                    raise error.TestFail(
+                        "Hotplug nic can't get ip after reboot vm")
+
+                error.context("Ping guest's new ip from host", logging.info)
+                status, output = ping_hotplug_nic(host_ip_addr, hotplug_nic["mac"],
+                                                  s_session, guest_is_linux)
+                if status:
+                    err_msg = "New nic failed ping test, error info: '%s'"
+                    raise error.TestFail(err_msg % output)
+
+                error.context("Pause vm", logging.info)
+                vm.monitor.cmd("stop")
```

Comment:
I think better  to pause VM  a long time(eg, 600s) then resume it and not pause it, then resume it immediately